### PR TITLE
fix: after navigating from new ds workflow to existing ds workflow, code no longer persists

### LINKED ISSUE #326: Workflow Steps Code persists when navigating from in-progress new workflow to existing dataset workflow
Observed just now in rosebud:

- Use "New Dataset", edit a cell
- Navigate to collection (accepting unsaved changes warning)
- Select a dataset with an existing workflow, navigate to its workflow route
- Cells reflect the code from the New Dataset

### PR DESCRIPTION
**Pull request by boandriy**
fixed issue where after changing code in new ds workflow you would navigate to existing ds workflow and cells would not change

Fixes #326

### CODE REVIEW

File: src/features/workflow/state/workflowState.ts [hunk 1]
```diff
@@ -289,6 +292,17 @@ function moveWorkflowTransformStepUp (state: WorkflowState, action: WorkflowStep
   return
 }
 
+function workflowUndoChanges (state: WorkflowState, actions: UndoWorkflowChanges) {
+  const oldTriggers:WorkflowTrigger[] | undefined = JSON.parse(JSON.stringify(state.workflowBase.triggers))
+  const oldHooks: WorkflowHook[] | undefined = JSON.parse(JSON.stringify(state.workflowBase.hooks))
+  const oldSteps: TransformStep[] = JSON.parse(JSON.stringify(state.workflowBase.steps || []))
```
Comment (b5) [inline src/features/workflow/state/workflowState.ts]: why do we need this double JSON parse/unparse?

File: src/features/workflow/state/workflowState.ts [hunk 2]
```diff
@@ -289,6 +292,17 @@ function moveWorkflowTransformStepUp (state: WorkflowState, action: WorkflowStep
...
+  state.workflow.triggers = oldTriggers
+  state.workflow.hooks = oldHooks
+  if(state.dataset.transform){
```
Comment (b5) [inline src/features/workflow/state/workflowState.ts]: if statement spacing

File: src/features/workflow/state/workflowState.ts (hunk 1 above)
Comment (boandriy) [reply src/features/workflow/state/workflowState.ts]: its a trick to make a deep copy of an object

**Review by ramfox — APPROVED** [review]
LGTM!

Before you merge, let's change the title of this PR to be more about the _effect_ of the change you are making, rather than _why_ you are making it. To say it another way, when we make bug fixes, the commits and PR titles we write should be about "what happens now" rather than "what was wrong before".

example:
`fix(workflow): do not keep code changes after clicking "discard changes"`

**Pull request merged**